csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -780,6 +783,62 @@ private static void HexEncodeUtf16CodeUnit(TextWriter writer, char c)
             writer.Write(Hex[(c >> 4) & 0xf]);
             writer.Write(Hex[(c >> 0) & 0xf]);
         }
+        
+        
+        private void WriteJsonObjectOpen(TextWriter writer, char openChar = '{')
```

Comment:
I see the point yes. So, I've tried to side-step this problem as follows:

- Renamed `WriteJsonObjectEntry` to `WritePropertySeparator` (this is consistent with the existing consts containing the actual separators). A similar argument can be made that list entries aren't really properties (or are they? javascript is weird). But since property already was there, it's at least consistent.
- Renamed `WriteJsonObjectOpen` to `WriteBracketOpen`. Calling this method each time with an actual bracket argument, for example  
  ```csharp
  WriteBracketOpen(writer, ObjectOpenBracket);
  // bool writtenFields = WriteMessageFields(writer, message, false, indentationLevel + 1);
  WriteBracketClose(writer, ObjectCloseBracket, writtenFields, indentationLevel);
  ```

So the commonality between `{}` and `[]` is that they are brackets, so we can avoid the difficulty of coming up with a term for their meaning.